Guide/Constructive/New Year Chaos/solution.cpp

```diff
@@ -0,0 +1,131 @@
+#include <iostream>
+#include <string>
+
+int main() {
+  int T; // Number of test cases     1 <= T <= 10
+  int N; // Number of people in line 1 <= N <= 10^5
+
+  std::cin >> T;
+
+  for (int t = 0; t < T; t++) { // For each test case
+    std::cin >> N;              // Get the number of people
+    int n;
+
+    /*
+       So we can have a 1-based array as opposed to 0-based, makes things easier
+          since n>=1
+     */
+    int line[N + 1];
+
+    /*
+       Flag here so we can check for chaotic while reading in the numbers to
+          prevent uneeded calculations in the future
+     */
+    bool is_chaotic = false;
+
+    /*
+       Read in the numbers and store them in the line array.
+       This is also where we check for chaotic or not. If by the end of this
+          loop we haven't said its chaotic, we never will.
+
+       Note: n=1 Because we are working with 1-based array instead of 0
+     */
+    for (n = 1; n <= N; n++) {
+      int x;
+      std::cin >> x;
+
+      /*
+         The reason for the flag instead of simply breaking here is because we
+            still need to read in all the inputs.
+         This condition is the ONLY way for the line to be chaotic.
+         If it is chaotic we don't need to waste the time to store the values
+       */
+      if (x > n + 2) is_chaotic = true;
+      else line[n] = x;
+    }
+
+    if (is_chaotic) {
+      /*
+         Now that we have read all the numbers in, we can call it quits now if
+            its chaotic
+       */
+      std::cout << "Too chaotic" << std::endl;
+    } else {
+      int count = 0; // AKA number of bribes
+
+      /*
+         Note we are iterating backwards and arranging people as we go.
+         This means that at any point, we can assume everybody behind that point
+            is where they are supposed to be.
+          This forces the situation where if
+            line[i] != i, then either line[i-1] = i or line[i-2] = i
+       */
+      for (n = N; n > 0; n--) {
+        /*
+           We only care about 3 people & 3 positions at a time.

```

Comment:
Eh no.  I should take this out.  The comments predate the Readme and are from when this file was a standalone. I'll leave comments that don't get covered by the readme